ckanext/harvest/logic/action/get.py

```diff
@@ -30,11 +30,23 @@ def harvest_source_show(context,data_dict):
 
     :param id: the id or name of the harvest source
     :type id: string
+    
+    :param url: url of the harvest source (as an alternative to the id)
+    :type url: string
 
     :returns: harvest source metadata
     :rtype: dictionary
     '''
+    model = context.get('model')
 
+    # Find the source by URL
+    if data_dict.get('url') and not data_dict.get('id'):
```

Comment:
Are these standard python `dict`s? Will this throw an error if the key does not exist, or are they guaranteed to exist?